lib/json/logger.ex

```diff
@@ -0,0 +1,49 @@
+defmodule Json.Logger do
+  @moduledoc """
+  Exposes separate log level configuration so developers can set logging
+  verbosity for json library
+
+  To configure log level only for json library, add folowing line in config file
+
+      use Mix.Config
+      # to make json be very verbose
+      config :json, log_level: :debug
+      # to make json be silent
+      config :json, log_level: :error
+  """
+  require Logger
+  @levels [:debug, :info, :warn, :error]
+  @level Application.get_env(:json, :log_level, :info)
+
+  @allowed_levels @levels
+                  |> Enum.reverse()
+                  |> Enum.reduce_while([], fn l, acc ->
+                    if l == @level do
+                      {:halt, [@level | acc]}
+                    else
+                      {:cont, [l | acc]}
+                    end
+                  end)
+                  |> Enum.reverse()
+
+  @doc false
+  @spec allowed_levels() :: [:debug | :error | :info | :warn, ...]
+  def allowed_levels(), do: @allowed_levels
+
+  @doc """
+  Logs given message to logger at given log level
+
+  Supported log levels are:
+  - `:debug` - All messages are logged
+  - `:info` - only :info, :warn and :error messages are logged
+  - `:warn` - only :warn and :error messages are logged
+  - `:error` - only :error messages are logged
+  """
+  defmacro log(level, message) do
```

Comment:
Maybe make wrap `message` into an anonymous function so that the message it self is only generated if needed? Cause doing some heavy IO.inspects can be nasty even if we discard the message at end, its still doing all the work of `inspect`ing a bunch of stuff. 

Then instead of:

```
Logger.log(level, message)
```
you'd do 

```
Logger.log(level, message.())
```

Reply:
What do you think?

Reply:
Not needed because Logger accept message to be function or string. So pass `fn -> "some text message" end` and it will be executed by Logger module.

Reply:
would you mind going through the log calls and wrap them in lambdas then, please? 

So people don't need to use `config :logger, compile_time_purge_level: :debug`

Reply:
done